unyt/array.py

```diff
@@ -516,15 +504,6 @@ def __new__(cls, input_array, input_units=None, registry=None, dtype=None,
             if registry is not None:
                 obj.units.registry = registry
             return obj
-        if input_array is NotImplemented:
-            return input_array.view(cls)
-        if registry is None and isinstance(input_units, (str, bytes)):
```

Comment:
Will we have a check for this once we start using `unyt` in `yt`?

Reply:
Yeah, I removed this because the error message is highly yt-specific. We could try to massage the error message, but then that's less useful for yt users.

Another way to do it would be to add the error in a wrapper around `unyt_array` in yt. Although that would mean we'd need to keep using that wrapper going forward instead of just exposing `unyt_array`.

I'm open to suggestions about how to do this. I'd prefer not to leave the error in as-is though because it only makes sense in the context of yt.

Reply:
Happy to just take the error out now--can you file an issue to remind us to fix this for yt-4.0?